stdlib/public/stubs/UnicodeNormalization.cpp

```diff
@@ -129,8 +129,8 @@ int32_t _swift_stdlib_unicode_compare_utf16_utf16(const uint16_t *LeftString,
                                                   const uint16_t *RightString,
                                                   int32_t RightLength) {
   return ucol_strcoll(GetRootCollator(),
-    LeftString, LeftLength,
-    RightString, RightLength);
+    (const UChar *)LeftString, LeftLength,
+    (const UChar *)RightString, RightLength);
```

Comment:
Thank you for checking, but this does not prove that the cast is correct.  In fact, clang does not warn on this code on OS X, where `sizeof(wchar_t) == 4`, even if I dereference the pointer with `printf("%d\n", *wc_var);`.

My interpretation of N4431 3.10/10 is that dereferencing that pointer would be a violation of strict aliasing.  @rjmccall?